### ISSUE TITLE
Race deduplication is based on LEA ID and not recency

### ISSUE THREAD

**Issue by jayckaiser** [edanalytics/edu_wh#129]
https://github.com/edanalytics/edu_wh/blob/1c7546e2487e5631f6970992ce516e1a1e7b133d/models/build/edfi_3/students/bld_ef3__choose_stu_demos.sql#L12

This arbitrarily chooses the record with the lower LEA ID and not the most recent record in a multi-tenant warehouse.

**Comment by ejoranlienea** [comment]
This is sort of true. A couple points of clarification:
- It's partitioned by `k_student`, so multi-tenant warehouses are fine, only warehouses that are single-tenant with multiple districts are affected by this. This is a rare configuration, and a challenging one to handle given the structure of Ed-Fi.
- The most recent record by ed org is chosen here, so there is a recency rule baked into this: https://github.com/edanalytics/edu_edfi_source/blob/main/models/staging/edfi_3/stage/stg_ef3__student_education_organization_associations.sql
- When there are multiple competing edorgs within a single tenant (which can only really occur in the rare case outlined in bullet 1), choosing by recency here would have the 'last writer' problem, where the winning record could flip-flop. The approach taken here achieves consistency across runs

**Comment by jayckaiser** [comment]
These are good points. We discovered this issue in the SCDE warehouse while discussing business rules for growth. We found that for a given student in two ed-orgs, the race was chosen based on alphabetic LEA ID, which really surprised the analysts.

Julia requests that we consider the "last writer" problem as a preferable and more justifiable solution in cases like these. This is an easier deduplication business rule to communicate to stakeholders over our more arbitrary alphabetical situation. Although the record may flip-flip over time, this would result in a more accurate representation of a student at a given point in time.
